ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto

```diff
@@ -111,15 +135,53 @@ message AllocatePartyRequest {
 
   // Human-readable name of the party to be added to the participant. It doesn't
   // have to be unique.
+  // Use of this field is discouraged. Consider using modifiable participant local party metadata instead.
   // Optional
   string display_name = 2;
+
+  // Participant local metadata to be bound to this newly allocated party.
+  // Optional
+  ObjectMeta metadata = 3;
 }
 
 message AllocatePartyResponse {
 
   PartyDetails party_details = 1;
 }
 
+// The participant local metadata of a party.
+// This metadata is stored locally on the backing participant and is not exposed on the given ledger.
+message PartyMetadata {
+
+  // The stable unique identifier of a Daml party.
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string party = 1;
+
+  // The metadata of this party.
+  // Optional
+  ObjectMeta metadata = 2;
+}
+
+// Required authorization: ``HasRight(ParticipantAdmin)``
+message UpdatePartyMetadataRequest {
```

Comment:
In a similar vein, perhaps this could be `UpdatePartyDetailsRequest`, and the `PartyMetadata` be dropped and replaced with an `ObjectMeta metadata` instead.

Reply:
It seems it makes sense to differentiate between on-ledger vs. participant-local resources. Party is both and so we have a duality of `PartyDetails` and `PartyMetadata`. It might be not ideal as it is now and would've looked differently if we were to redesign the party API from scratch.

Reply:
Are you suggesting that `UserDetails` and `UserMetadata` types might be a better design? I'm indifferent to which approach is taken, but I don't see a reason why they should be different other than for historical reasons, and even then it's not really historical since the idea of metadata is being added to both `User`'s and `Party`'s at the same time.

Reply:
For the record, we're going for an `UpdateUserDetails` rpc which operates on `UserDetails` message